From ae4d76983de072191fc451763e04b502357f2022 Mon Sep 17 00:00:00 2001
From: Yann Gautier <yann.gautier@foss.st.com>
Date: Thu, 11 Jan 2024 19:34:24 +0100
Subject: [PATCH 444/841] fix(stm32mp2): add INIT_UNUSED_NS_EL2 flag

As STM32MP2 does not boot BL33 at EL2, we need to initialize the unused
EL2 registers with INIT_UNUSED_NS_EL2 flag.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git b78b07f9c93d82234298b206ffb1a591d00b41b2 fix(stm32mp2): add INIT_UNUSED_NS_EL2 flag]
Change-Id: Ic6f65c931752fc7c2ed9043500edb17cdd43b3c4
Signed-off-by: Yann Gautier <yann.gautier@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/350822
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 plat/st/stm32mp2/platform.mk | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/plat/st/stm32mp2/platform.mk b/plat/st/stm32mp2/platform.mk
index 41590a0fd..b7e644550 100644
--- a/plat/st/stm32mp2/platform.mk
+++ b/plat/st/stm32mp2/platform.mk
@@ -1,5 +1,5 @@
 #
-# Copyright (c) 2023, STMicroelectronics - All Rights Reserved
+# Copyright (c) 2023-2024, STMicroelectronics - All Rights Reserved
 #
 # SPDX-License-Identifier: BSD-3-Clause
 #
@@ -14,6 +14,7 @@ CRASH_REPORTING			:=	1
 ENABLE_PIE			:=	1
 PROGRAMMABLE_RESET_ADDRESS	:=	1
 BL2_IN_XIP_MEM			:=	1
+INIT_UNUSED_NS_EL2		:=	1
 
 # Disable features unsupported in ARMv8.0
 ENABLE_SPE_FOR_NS		:= 0
-- 
2.47.2

